salt/grains/core.py

```diff
@@ -1668,6 +1669,22 @@ def append_domain():
         grain['append_domain'] = __opts__['append_domain']
     return grain
 
+def _read_all_fqdns():
+    fqdns = set()
+    for ip_addr in salt.utils.network.ip_addrs(include_loopback=False):
```

Comment:
Could `ip_addrs` be retrieved from some `grains['key']` so that the cached value is reused instead of recomputing it?

Reply:
AFAIU it's not possible to access Salt's grains inside grains.

Reply:
Here that is done: https://github.com/saltstack/salt/pull/45060/files#diff-6e894b39b67c4bab67796752744ea9d5L1653

Reply:
Fixed